compiler/testData/diagnostics/tests/controlFlowAnalysis/unnecessaryLateinit/lateinitWithPlusAssign.fir.kt

```diff
@@ -0,0 +1,8 @@
+class Foo {
+    <!UNNECESSARY_LATEINIT!>lateinit<!> var bar: String
+
+    constructor(baz: Int) {
+        // At best, we should have error here despite of lateinit
+        bar += baz
```

Comment:
This is the regression I mentioned at the PR description. A quick fix came to me, but it's a bit complicated than expected. Not sure how long it will take or can fit in this PR. Since it's a warning, I think it's okay to add that temporarily.

Reply:
For FE1.0 this results in an error at runtime when you do `Foo(1).bar`: `Caused by: kotlin.UninitializedPropertyAccessException: lateinit property bar has not been initialized`. It would be nice to catch this error but that's out of scope here.

Reply:
Yeah, noticed that as well. Added a note to Priority 2B to not forget.